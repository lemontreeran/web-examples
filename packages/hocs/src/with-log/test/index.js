/* eslint-disable no-console */
import React from 'react'
import { mount } from 'enzyme'

import withLog from '../src/'

const Target = () => null

describe('withLog', () => {
  beforeAll(() => {
    jest.spyOn(console, 'log').mockImplementation(() => {})
  })

  afterEach(() => {
    console.log.mockClear()
  })

  afterAll(() => {
    console.log.mockRestore()
  })

  it('should log all props if no argument was passed in', () => {
    const EnhancedTarget = withLog()(Target)

    mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(console.log.mock.calls).toMatchSnapshot()
  })

  it('should call provided `getMessageToLog` with props and log its result', () => {
    const mockGetMessageToLog = jest.fn(({ a }) => `a = ${a}`)
    const EnhancedTarget = withLog(mockGetMessageToLog)(Target)

    mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(mockGetMessageToLog.mock.calls).toMatchSnapshot()
    expect(console.log.mock.calls).toMatchSnapshot()
  })

  describe('env', () => {
    const origNodeEnv = process.env.NODE_ENV

    afterAll(() => {
      process.env.NODE_ENV = origNodeEnv
    })

    it('should pass Target through in production env', () => {
      process.env.NODE_ENV = 'production'

      const EnhancedTarget = withLog()(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should wrap display name in non-production env', () => {
      process.env.NODE_ENV = 'test'

      const EnhancedTarget = withLog()(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
