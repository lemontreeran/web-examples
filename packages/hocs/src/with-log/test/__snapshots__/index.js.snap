// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLog env should pass Target through in production env 1`] = `<Target />`;

exports[`withLog env should wrap display name in non-production env 1`] = `
<withLog(Target)>
  <Target />
</withLog(Target)>
`;

exports[`withLog should call provided \`getMessageToLog\` with props and log its result 1`] = `
Array [
  Array [
    Object {
      "a": 1,
      "b": 2,
      "c": 3,
    },
  ],
]
`;

exports[`withLog should call provided \`getMessageToLog\` with props and log its result 2`] = `
Array [
  Array [
    "withLog(Target):",
    "a = 1",
  ],
]
`;

exports[`withLog should log all props if no argument was passed in 1`] = `
Array [
  Array [
    "withLog(Target):",
    Object {
      "a": 1,
      "b": 2,
      "c": 3,
    },
  ],
]
`;
